### PR TITLE
chore(sucrase)!: upgrade sucrase dependency

### PR DESCRIPTION
**Pull request by curran**
## Rollup Plugin Name: `sucrase`

This PR contains:

- [ ] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [X] other

Are tests included?

- [ ] yes (_bugfixes and features will not be merged without tests_)
- [X] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [X] no

If yes, then include "BREAKING CHANGES:" in the first commit message body, followed by a description of what is breaking. 

List any relevant issue numbers:

Closes https://github.com/rollup/plugins/issues/936
Closes https://github.com/rollup/plugins/issues/937

### Description

Greetings, I noticed that the Sucrase plugin dependencies are a bit stale. I figured I'd submit a PR to bump them. This was automated using [npm-check-updates](https://www.npmjs.com/package/npm-check-updates). I also updated the test fixtures.

### CODE REVIEW

File: packages/sucrase/package.json [hunk 1]
```diff
@@ -44,14 +44,14 @@
     "jsx"
   ],
   "peerDependencies": {
-    "rollup": "^1.20.0 || ^2.0.0"
+    "rollup": "^2.53.1"
```
Comment (curran) [inline packages/sucrase/package.json:47]: We might want to keep the `||` to support older Rollup? Happy to change this line back to what it was if that would be preferable.

Comment (shellscape) [reply packages/sucrase/package.json:47]: I'm good with this change. We'll do a new major version.

**Comment by shellscape** [comment]
You're going to have to run `pnpm install` and commit the lockfile update for CI to pass.

**Comment by curran** [comment]
Thanks @shellscape for reviewing!

I updated from upstream and committed `pnpm-lock.yaml`.

Also upgraded dependencies again. Some have changed in the past 2 weeks.

**Comment by shellscape** [comment]
thanks!

**Comment by curran** [comment]
Thanks @shellscape !

**Pull request merged**